doc/cpp-style-guide.md

```diff
@@ -1,629 +1,61 @@
 # C++ Style Guide for google-cloud-cpp
 
-This project largely follows (and extensively copies) the [Google Style Guide][google-style-guide-link] (hereafter GSG).
-The exceptions are highlighted to help contributors familiar with the GSG start quickly.
+This project follows the [Google Style Guide][google-style-guide-link]
+(hereafter GSG). In a few cases where the GSG gives latitude, we have made a
+choice for consistency within this project. The following describes where we
+differ from the GSG.
 
 [google-style-guide-link]: https://google.github.io/styleguide/cppguide.html
 
-## Goals
+## Reference Arguments
 
-##### Style rules should pull their weight
+Function return values are preferred over argument out-params. However, when an
+out-param is appropriate, prefer a non-const reference.
 
-The benefit of a style rule must be large enough to justify asking all of our engineers to remember it. The benefit is
-measured relative to the codebase we would get without the rule, so a rule against a very harmful practice may still
-have a small benefit if people are unlikely to do it anyway. This principle mostly explains the rules we don’t have,
-rather than the rules we do: for example, goto contravenes many of the following principles, but is already vanishingly
-rare, so the Style Guide doesn’t discuss it.
+As of this writing, this rule violates the GSG.
 
-##### Optimize for the user first, the reader second, and the writer last
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Reference_Arguments)
```

Comment:
I would use a more descriptive link name than "link to GSG". Something like "GSG's section on Reference Arguments". Bad link names are [not good for accessibility](https://www.accessibilityoz.com/2014/02/links-and-accessibility/).

Reply:
Accessibility is good. Done.